File: SDK-2.4.4/iOS/samples/sdk-ios-sample-objc/FIS Digital Payments SDK/ViewController.m
//
//  ViewController.m
//  FIS Digital Payments SDK
//
//  Created by Zimmerman, Maxwell on 5/2/19.
//  Copyright © 2019 FIS. All rights reserved.
//

#import "ViewController.h"
#import "WebKit/WKWebView.h"

@interface ViewController ()

@end

@implementation ViewController

WKWebView *webView;
NSArray *domainWhitelist = nil;

+ (void) initialize {
    domainWhitelist = @[@"epayments-crossdomain-fnc.billdomain.com", @"epayments-epayui-fnc-3.money-movement.com", @"epayments-epayui-uat-3.money-movement.com"];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    webView = [[WKWebView alloc]initWithFrame:CGRectZero];
    [FDSSDKManager createWithWebView:webView domainWhitelist:domainWhitelist webEventListener:self];

    self.view = webView;
    
    NSURL *url = [NSURL URLWithString: @"https://epayments-crossdomain-fnc.billdomain.com/Mobile/Index.html"];
    NSURLRequest *urlRequest = [NSURLRequest requestWithURL:url];
    
    [webView loadRequest:urlRequest];
}

- (void)onEventReceivedWithWebEvent:(FDSWebEvent * _Nonnull)webEvent {
    printf("Event received.");
}

@end

File: SDK-2.4.4/iOS/samples/sdk-ios-sample-swift/FISDigitalPaymentsSDK Sample/ViewController.swift
//  ViewController.swift
//  FISDigitalPaymentsSDK Sample
//  Created by Zimmerman, Maxwell on 2/13/19.
//  Copyright © 2019 FIS. All rights reserved.
//

import FISDigitalPaymentsSDK
import WebKit
import UIKit

class ViewController: UIViewController, FDSWebEventListening {
    let domainWhitelist = ["epayments-crossdomain-fnc.billdomain.com", "epayments-epayui-fnc-3.money-movement.com", "epayments-epayui-uat-3.money-movement.com"]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let webView = WKWebView()
        view = webView
        
        _ = FDSSDKManager.create(webView: webView, domainWhitelist: domainWhitelist, webEventListener: self)
        
        webView.load(URLRequest(url: URL(string: "https://epayments-crossdomain-fnc.billdomain.com/Mobile/Index.html")!))
    }
    
    func onEventReceived(webEvent: FDSWebEvent) {
        print(webEvent)
    }
}
